cpp/src/arrow/compute/exec/expression_benchmark.cc

```diff
@@ -69,6 +70,26 @@ static void SimplifyFilterWithGuarantee(benchmark::State& state, Expression filt
   }
 }
 
+static void ExecuteScalarExpressionOverhead(benchmark::State& state, Expression expr) {
+  const auto rows_per_batch = static_cast<int32_t>(state.range(0));
+  const auto num_batches = 10000000 / rows_per_batch;
+
+  ExecContext ctx;
+  auto dataset_schema = schema({
+      field("x", int64()),
+  });
+  ExecBatch input({Datum(ConstantArrayGenerator::Int64(rows_per_batch, 5))},
+                  /*length=*/1);
```

Comment:
Hmm, shouldn't this be `/*length=*/rows_per_batch` (it may not actually matter, depending on the expression being executed)?

Reply:
I think the length should be `rows_per_batch`, but it's interesting to see that with length 1 too, this worked, maybe because we have an Array in the schema and not a scalar value?